### PR TITLE
Adding section for protect function

### PR DESCRIPTION
**Pull request by dockerymick**
https://app.zenhub.com/workspaces/engineering-backlog-do-not-delete-604fab62d4b98d00150a2854/issues/stolostron/backlog/19357

### CODE REVIEW

File: governance/custom_template.adoc [hunk 1]
```diff
@@ -393,6 +394,37 @@ spec:
         {{ (fromConfigMap "site-config" "site1" "enabled")  | toBool }}
 ----
 
+[#protect-function]
+=== protect function
+
+The `protect` function enables you to encrypt a string in a hub cluster policy template. View the following example of the configuration policy that uses the `protect` function:
```
Comment (mprahl) [inline governance/custom_template.adoc]:
```suggestion
The `protect` function enables you to encrypt a string in a hub cluster policy template. It is automatically decrypted on the managed cluster when the policy is evaluated. View the following example of the configuration policy that uses the `protect` function:
```

File: governance/custom_template.adoc [hunk 2]
```diff
@@ -429,7 +465,7 @@ See the following table for a comparison of hub cluster and managed cluster temp
 | {{ … }}
 
 | Functions
```
Comment (mprahl) [inline governance/custom_template.adoc:467]: Could you please document that the `fromSecret` template function on the Hub will store the resulting value as an encrypted string on the replicated policy in the managed cluster namespace?

Essentially, it's the equivalent of calling `{{hub "(lookup "v1" "Secret" "default" "my-hub-secret").data.message | protect hub}}`.

File: governance/custom_template.adoc [hunk 3]
```diff
@@ -393,6 +394,37 @@ spec:
         {{ (fromConfigMap "site-config" "site1" "enabled")  | toBool }}
 ----
 
+[#protect-function]
```
Comment (mprahl) [inline governance/custom_template.adoc:397]: Could you please document that the encryption algorithm used is AES-CBC using 256 bit keys? Each encryption key is unique per managed cluster and is automatically rotated every 30 days. To force an immediate rotation, delete the `policy.open-cluster-management.io/last-rotated` annotation on the `policy-encryption-key` Secret in the managed cluster namespace on the Hub.

**Review by mprahl — CHANGES_REQUESTED** [review]

**Comment by dockerymick** [comment]
@mprahl Thanks for your feedback Matt, when you get an opportunity. May you re-review the PR?

File: governance/custom_template.adoc [hunk 4]
```diff
@@ -393,6 +394,41 @@ spec:
...
+        {{hub "(lookup "v1" "Secret" "default" "my-hub-secret").data.message | protect hub}}
+----
+
+In the previous YAML example, there is an existing hub cluster policy template that is defined to use the `lookup` function. On the replicated policy in the managed cluster namespace, the value might resemble the following syntax: `$ocm_encrypted:okrrBqt72oI+3WT/0vxeI3vGa+wpLD7Z0ZxFMLvL204=`
+
+Each encryption algorithm used is AES-CBC using 256-bit keys. Each encryption key is unique per managed cluster and is automatically rotated every 30 days.
+
+This ensures that your decrypted value is to never be stored in the policy on the managed cluster.
+
+To force an immediate rotation, delete the `policy.open-cluster-management.io/last-rotated` annotation on the `policy-encryption-key` Secret in the managed cluster namespace on the hub cluster. Policies are reprocessed to use the new encryption key.
```
Comment (mprahl) [inline governance/custom_template.adoc]:
```suggestion
To force an immediate rotation, delete the `policy.open-cluster-management.io/last-rotated` annotation on the `policy-encryption-key` Secret in the managed cluster namespace on the hub cluster. Policies are then reprocessed to use the new encryption key. 
```

File: release_notes/whats_new.adoc [hunk 5]
```diff
@@ -60,6 +60,8 @@ For other Application topics, see link:../applications/app_management_overview.a
 
 * Use the optional YAML field, `metadataComplianceType`, to process labels and annotations of an object differently than the other fields. For more information, see the link:../apis/policy.adoc#policy-api[Policy API].
 
+* You can use the `protect` function to secure sensitive data that is on your hub cluster policy template. See link:../governance/custom_template.adoc##protect-function[protect function] section for more information
```
Comment (mprahl) [inline release_notes/whats_new.adoc]: Could you also please mention that `fromSecret` is available in Hub policy templates?

**Comment by mprahl** [comment]
@dockerymick this looks great! I just left a couple of minor comments. After that is addressed, this can be merged from my perspective.

**Comment by dockerymick** [comment]
Thanks Matt!

**Comment by dockerymick** [comment]
@ckandag When you have an opportunity may you review as well?

**Review by ckandag — APPROVED** [review]

**Comment by dockerymick** [comment]
Thanks for your review @ckandag ! Moving forward and merging this PR

**Pull request merged**